client/lib/fifo/fifo_windows.go

```diff
@@ -31,6 +31,7 @@ func (f *winFIFO) Read(p []byte) (n int, err error) {
 
 		f.conn = c
 	}
+	f.connLock.Unlock()
```

Comment:
The pattern of using `f.connLock` for synchronizing some `f.conn` accesses but not others feel a bit odd and raises a yellow flag even if it's correct.  I'm not sure if our tests run these paths in goroutines for `-race` to pick it up and reading memory model[1] doesn't fell me with confidence about this.

I would suggest setting a local variable to `conn := f.conn` inside the lock and using it in rest of function or comment why this pattern is valid; if for nothing else to ease future reviewer of code.

[1] https://golang.org/ref/mem#tmp_8